kubernetes_env/benchmark/benchmark.py

```diff
@@ -24,19 +26,86 @@
 FORTIO_DIR = DIRS[2].joinpath("bin/fortio")
 
 
-def plot(xaxis_data, yaxis_data, xaxis_label, yaxis_label, query, graph_name):
-    plt.plot(xaxis_data, yaxis_data, marker='o')
-    plt.xlabel(xaxis_label)
-    plt.ylabel(yaxis_label)
-    plt.title(query)
-    util.check_dir(GRAPHS_DIR)
-    plt.savefig(str(GRAPHS_DIR.joinpath(f"{graph_name}.png")))
+def sec_to_ms(res_time):
+    return round(float(res_time) * 1000, 3)
 
 
-def run_fortio(platform, threads, qps, run_time):
+def convert_data(data):
+    return {
+        "Count":
+        data["Count"],
+        "Min":
+        sec_to_ms(data["Min"]),
+        "Max":
+        sec_to_ms(data["Max"]),
+        "Sum":
+        sec_to_ms(data["Sum"]),
+        "Avg":
+        sec_to_ms(data["Avg"]),
+        "StdDev":
+        sec_to_ms(data["StdDev"]),
+        "Data": [{
+            "Start": sec_to_ms(datum["Start"]),
+            "End": sec_to_ms(datum["End"]),
+            "Percent": datum["Percent"],
+            "Count": datum["Count"]
+        } for datum in data["Data"]]
+    }
+
+
+def plot(files):
+    fs = []
+    try:
+        names = []
+        for f in files:
+            names.append(f.replace(".json", ""))
+            fs.append(open(str(DATA_DIR.joinpath(f)), 'r'))
+        json_data = []
+        for f in fs:
+            json_data.append(json.load(f))
+        converted_durations = [
+            convert_data(data["DurationHistogram"]) for data in json_data
+        ]
+        dfs = []
+        title = ""
+        for (idx, converted_duration) in enumerate(converted_durations):
+            res_times = [0]
+            percentiles = [0]
+            avg = converted_duration["Avg"]
+            title += f"{names[idx]} Avg: {avg} ms"
+            for datum in converted_duration["Data"]:
+                res_times.append(datum["Start"])
+                res_times.append(datum["End"])
+                percentiles.append(float(datum["Percent"]))
+                percentiles.append(float(datum["Percent"]))
+                d = {
+                    "Response time (ms)": res_times,
+                    "Percentiles": percentiles,
+                    "Filter": names[idx]
+                }
+                df = pd.DataFrame(data=d)
+                dfs.append(df)
+        final_df = pd.concat(dfs)
+        fig, ax = plt.subplots(figsize=(12, 6))
+        sns.lineplot(data=final_df,
+                     x="Response time (ms)",
+                     y="Percentiles",
+                     hue="Filter",
+                     ci=False,
+                     marker='o')
+        plt.title(title)
+        plot_name = "-".join(names)
+        util.check_dir(GRAPHS_DIR)
+        plt.savefig(str(GRAPHS_DIR.joinpath(f"{plot_name}.png")))
```

Comment:
For concatenation like this I would recommend straight f-strings instead of joinpath. 
`f"{GRAPHS_DIR}/{plot_name}.png"`